### PR TITLE
Tracking Improvement

### PR DESCRIPTION
**Pull request by lily1128**

### CODE REVIEW

File: app/src/js/common/track_policies/polygon_linear_interpolation_policy.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    this._policyType = types.TrackPolicyType.LINEAR_INTERPOLATION_POLYGON
+  }
+
+  /**
+   * Callback for when a label in the track is updated
+   */
+  public onLabelUpdated (
+
+  ) {
+    return 0
```
Comment (gycn) [inline app/src/js/common/track_policies/polygon_linear_interpolation_policy.ts]: I think you can change this to `return` without the `0`

File: app/src/js/server/create_project.ts [hunk 2]
```diff
@@ -221,7 +221,14 @@ export function createProject (
...
   const bundleFile = util.getBundleFile(form.labelType)
-  const [itemType, tracking] = util.getTracking(form.itemType)
+  const [itemType, tracking, policyTypes] =
+  util.getTracking(form.itemType, form.labelType)
+  let policy: string[]
+  if (tracking === true && policyTypes !== 'None') {
+    policy = [policyTypes]
+  } else {
+    policy = []
+  }
```
Comment (gycn) [inline app/src/js/server/create_project.ts]: I believe you do not need to check whether it is tracking and you can always just pass in the policy types. The front-end will check whether or not tracking is enabled and decide whether to use the policy types

Comment (lily1128) [reply app/src/js/server/create_project.ts]: I modified the getTracking function in util, which does not check whether it is tracking. Should I change the checking of whether it is tracking and return the policy type to be in that function?

File: app/src/js/common/track_policies/polygon_linear_interpolation_policy.ts (hunk 1 above)
Comment (lily1128) [reply app/src/js/common/track_policies/polygon_linear_interpolation_policy.ts]: Changed it~

File: app/src/js/common/track_policies/box2d_linear_interpolation_policy.ts [hunk 3]
```diff
@@ -163,6 +163,23 @@ function linearInterpolateBox2D (
     )
   }
 
+  if (nextManualIndex === -1) {
```
Comment (gycn) [inline app/src/js/common/track_policies/box2d_linear_interpolation_policy.ts]: Can you change this to an else if statement with the last if statement?

File: app/src/js/common/track_policies/box3d_linear_interpolation_policy.ts [hunk 4]
```diff
@@ -165,6 +165,22 @@ function linearInterpolateBox3D (
     )
   }
 
+  if (nextManualIndex === -1) {
```
Comment (gycn) [inline app/src/js/common/track_policies/box3d_linear_interpolation_policy.ts]: Could you also make this an else if?

File: app/src/js/server/create_project.ts (hunk 2 above)
Comment (gycn) [reply app/src/js/server/create_project.ts]: Could you change this according to what we discussed last week?

**Pull request merged**